File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int n = needle.length();
        for (int i = 0; i < haystack.length(); i++){
            if (haystack[i] == needle[0]){
                string check = haystack.substr(i,n);
                if (check == needle){
                    return i;
                }
            }
        }
        return -1;
    }
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        # if len(needle) > len(haystack):
        #     return -1

        # index = -1
        # matches = False
        # for i in range(len(haystack)): 
        #     j = 0
        #     if haystack[i] == needle[j]:
        #         matches = True
        #         index = i
        #         while matches and j < len(needle):
        #             if haystack[i+j] != needle[j]:
        #                 index = -1
        #                 matches = False
        #             j += 1
        #         if matches != False: 
        #             break
        
        # return index

        return haystack.find(needle)
        